src/main/java/carbon/task/Deadline.java

```diff
@@ -0,0 +1,133 @@
+package carbon.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.Temporal;
+
+import carbon.error.CarbonException;
+import carbon.error.InvalidFlagException;
+import carbon.error.InvalidParamException;
+import carbon.error.InvalidTimeException;
+
+/**
+ * Deadline class that represents tasks to be completed before a particular date or time.
+ * Can be represented with either date alone, or date together with time.
+ */
+public class Deadline extends Task {
+    public static final String FLAG = " /by";
+    private static final int TYPEKEY = Task.getTypeKey(Task.Type.DEADLINE);
+
+    private Temporal dateTime;
+
+    private Deadline(String name, Boolean isDone, Temporal dateTime) {
+        super(name, isDone);
+        this.dateTime = dateTime;
+    }
+
+    private static String extractName(String input) throws CarbonException {
+        int flagIndex = input.indexOf(Deadline.FLAG);
+        if (flagIndex == -1) {
+            CarbonException invalidFlag = new InvalidFlagException(input, Task.Type.DEADLINE);
+            throw invalidFlag;
+        } else {
+            String name = input.substring("deadline ".length(), flagIndex);
+            return name;
+        }
+    }
+
+    private static Temporal extractTime(String input) throws CarbonException {
+        int len = input.length();
+        int flagIndex = input.indexOf(Deadline.FLAG);
+        if (len <= flagIndex + Deadline.FLAG.length() + 1) {
+            CarbonException invalidParam = new InvalidParamException(input);
+            throw invalidParam;
+        } else {
+            String timeString = input.substring(flagIndex + Deadline.FLAG.length() + 1);
+            try {
+                Temporal dateTime = Deadline.formatTime(timeString);
+                return dateTime;
+            } catch (DateTimeParseException error) {
+                CarbonException invalidTime = new InvalidTimeException(timeString);
+                throw invalidTime;
+            }
+        }
+    }
```

Comment:
Perhaps it may be better to store the value of flagIndex + Deadline.FLAG.length() + 1 into a variable so that the code will be more readable.